From 3302c00f92c2597a08a0f30247b8011071f64aa7 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Thu, 29 Sep 2022 13:18:10 +0000
Subject: [PATCH] GCC: fix ambigous call to base::to_address

GCC sees std::to_address and base::to_address in C++20 mode.
---
 base/cxx20_to_address.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/base/cxx20_to_address.h b/base/cxx20_to_address.h
index 23b66e3..0010aec 100644
--- a/base/cxx20_to_address.h
+++ b/base/cxx20_to_address.h
@@ -30,7 +30,7 @@ constexpr auto to_address(const Ptr& p) noexcept
 
 template <typename Ptr, typename... None>
 constexpr auto to_address(const Ptr& p, None...) noexcept {
-  return to_address(p.operator->());
+  return base::to_address(p.operator->());
 }
 
 }  // namespace base
-- 
2.35.1

